### PR TITLE
fix: getMonth compatibility date

### LINKED ISSUE #1766: 🐛 BUG: suggested `compatibility_date` when publishing has month off-by-one
### What version of `Wrangler` are you using?

2.0.28

### What operating system are you using?

Mac

### Describe the Bug

1. Run `wrangler publish` without `compatibility_date` set in `wrangler.toml`
2. Suggested `compatibility_date` will be for previous month

### Erroneous Code

[`getMonth()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getMonth#return_value) returns an integer from `0` to `11` inclusive.

https://github.com/cloudflare/wrangler2/blob/27ad80eed7f25393a0e5c1d8a62c3b0e743a639d/packages/wrangler/src/publish.ts#L225

### PR DESCRIPTION
**Pull request by JacobMGEvans**
The getMonth() return values 0-11, adding 1 resolves this and added test.

resolves #1766

### CODE REVIEW

File: .changeset/rude-seas-impress.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"wrangler": patch
+---
+
+fix: getMonth compatibility date
```
Comment (petebacondarwin) [inline .changeset/rude-seas-impress.md:5]: We should write these messages from the point of view of a Wrangler user reading the CHANGELOG. I am not sure that "getMonth compatibility date" would mean much to them.

How about:

```suggestion
fix: set correct month for `compatibility_date` when initializing a new Worker
```

File: packages/wrangler/src/__tests__/publish.test.ts [hunk 2]
```diff
@@ -3233,6 +3233,29 @@ addEventListener('fetch', event => {});`
 		      `);
 		});
 
+		it("should error if a compatibility_date and suggest the correct month", async () => {
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/publish.test.ts:3236]: ??

```suggestion
		it("should error if a compatibility_date is missing and suggest the correct month", async () => {
```

File: packages/wrangler/src/publish.ts [hunk 3]
```diff
@@ -222,7 +222,9 @@ export default async function publish(props: Props): Promise<void> {
 
 	if (!(props.compatibilityDate || config.compatibility_date)) {
 		const compatibilityDateStr = `${new Date().getFullYear()}-${(
-			new Date().getMonth() + ""
+			new Date().getMonth() +
+			1 +
+			""
```
Comment (petebacondarwin) [inline packages/wrangler/src/publish.ts:227]: Out of curiosity, any idea why these `""` are here?
Perhaps they were previously there when the string was being built via concatenation? And now we are using string interpolation they are not needed, right?

**Review by petebacondarwin — APPROVED** [review]
Approving but I would love to see the changeset improved.

File: packages/wrangler/src/publish.ts (hunk 3 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/publish.ts:227]: It seems to have been a hack to coerce the type into a string for the following `.padStart(2, "0")` which could do this `String(new Date().getMonth() + 1)` instead which is also far more explicit of what/why its being done

Comment (JacobMGEvans) [reply packages/wrangler/src/publish.ts:227]: > It seems to have been a hack to coerce the type into a string for the following `.padStart(2, "0")` which could do this `String(new Date().getMonth() + 1)` instead which is also far more explicit of what/why its being done

This actually seems to change the output just slightly enough to effect the tests.

**Pull request merged**